vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/crawler/LinkCrawler.java

```diff
@@ -69,14 +69,20 @@ public void visit(Page page)
         }
     }
 
-    private static boolean isAllowedUrl(Page referringPage, WebURL url)
+    private boolean isAllowedUrl(Page referringPage, WebURL url)
     {
-        return isNotFile(url) && isFromTheSameSite(referringPage.getWebURL().getURL(), url.getURL());
+        return isUrlHasAllowedExtension(url) && isFromTheSameSite(referringPage.getWebURL().getURL(), url.getURL());
     }
 
-    private static boolean isNotFile(WebURL url)
+    private boolean isUrlHasAllowedExtension(WebURL url)
     {
-        return !StringUtils.endsWithAny(url.getPath().toLowerCase(), EXCLUDED_EXTENSIONS);
+        String fileExtensionFromUrl = FileNameUtils.getExtension(url.getPath().toLowerCase());
+        boolean allowed = fileExtensionFromUrl.isEmpty() || allowedExtensions.contains(fileExtensionFromUrl);
```

Comment:
users may write property value in upper-case, so it's better to convert to lower-case from both sides: the actual extension from URL and the values in the property